apps/accounts/models/provider_request.py

```diff
@@ -0,0 +1,111 @@
+    OPEN = "Open"  # GWF staff requested some changes from the provider
+
+
+class ProviderRequest(TimeStampedModel):
+    name = models.CharField(max_length=255)
+    website = models.CharField(max_length=255)
+    description = models.TextField()
+    status = models.CharField(choices=ProviderRequestStatus.choices, max_length=255)
+    created_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL, on_delete=models.CASCADE, null=True
+    )
+
+    def __str__(self):
+        return f"{self.name}"
+
+    def get_absolute_url(self):
+        return reverse("provider_request_detail", args=[str(self.id)])
+
+
+class ProviderRequestSupplier(models.Model):
+    # TODO: add filtering and/or validation to check
+    # that the selected services are offered by the selected supplier.
+
+    supplier = models.ForeignKey(Hostingprovider, on_delete=models.CASCADE)
+    services = TaggableManager(
+        verbose_name="Services used",
+        help_text="Click the services that your organisation uses from the selected supplier. These will be listed in the green web directory.",
+        blank=True,
+    )
+    request = models.ForeignKey(
+        ProviderRequest, on_delete=models.CASCADE, related_name="suppliers"
+    )
+
+    def __str__(self):
+        return f"{self.supplier}"
+
+
+class ProviderRequestLocation(models.Model):
+    city = models.CharField(max_length=255)
+    country = CountryField()
+    services = TaggableManager(
+        verbose_name="Services offered",
+        help_text="Click the services that your organisation offers. These will be listed in the green web directory.",
+        blank=True,
+    )
+    request = models.ForeignKey(ProviderRequest, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return f"{self.request.name} | {self.country}/{self.city}"
+
+
+class ProviderRequestASN(models.Model):
+    asn = models.IntegerField()
+    location = models.ForeignKey(ProviderRequestLocation, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return f"{self.asn}"
+
+
+class ProviderRequestIPRange(models.Model):
+    start = IpAddressField()
+    end = IpAddressField()
+    location = models.ForeignKey(ProviderRequestLocation, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return f"{self.start} - {self.end}"
+
+    def clean(self):
+        validate_ip_range(self.start, self.end)
+
+
+class EvidenceType(models.TextChoices):
+    ANNUAL_REPORT = "Annual report"
+    WEB_PAGE = "Web page"
+    CERTIFICATE = "Certificate"
+
+
+class ProviderRequestEvidence(models.Model):
+    title = models.CharField(max_length=255)
+    link = models.URLField(null=True, blank=True)
+    file = models.FileField(null=True, blank=True)
+    location = models.ForeignKey(ProviderRequestLocation, on_delete=models.CASCADE)
+    type = models.CharField(choices=EvidenceType.choices, max_length=255)
+
+    def __str__(self):
+        return f"{self.title} ({self.type})"
+
+    def clean(self):
```

Comment:
Could we have some tests on this logic please? It's 'our' code, and I don't think we have tests on the earlier implementation of this logic.